espflash/src/connection.rs

```diff
@@ -119,43 +125,77 @@ impl Connection {
-            self.serial.write_request_to_send(false)?;
+    fn classic_reset(&mut self, extra_delay: bool) -> Result<(), Error> {
+        info!(
+            "Attempting Classic reset with {} delay...",
+            if extra_delay { "extra" } else { "default" }
+        );
 
-            sleep(Duration::from_millis(100));
+        self.serial.write_data_terminal_ready(false)?;
+        self.serial.write_request_to_send(false)?;
 
-            self.serial.write_data_terminal_ready(true)?;
-            self.serial.write_request_to_send(false)?;
+        self.serial.write_data_terminal_ready(true)?;
+        self.serial.write_request_to_send(true)?;
 
-            sleep(Duration::from_millis(100));
+        self.serial.write_data_terminal_ready(false)?;
+        self.serial.write_request_to_send(true)?;
 
-            self.serial.write_request_to_send(true)?;
-            self.serial.write_data_terminal_ready(false)?;
-            self.serial.write_request_to_send(true)?;
+        sleep(Duration::from_millis(100));
 
-            sleep(Duration::from_millis(100));
+        self.serial.write_data_terminal_ready(true)?;
+        self.serial.write_request_to_send(false)?;
 
-            self.serial.write_data_terminal_ready(false)?;
-            self.serial.write_request_to_send(false)?;
-        } else {
-            self.serial.write_data_terminal_ready(false)?;
-            self.serial.write_request_to_send(true)?;
+        let millis = if extra_delay { 500 } else { 50 };
+        sleep(Duration::from_millis(millis));
 
-            sleep(Duration::from_millis(100));
+        self.serial.write_data_terminal_ready(false)?;
+        self.wait_for_connection()
+    }
 
-            self.serial.write_data_terminal_ready(true)?;
-            self.serial.write_request_to_send(false)?;
+    #[cfg(unix)]
+    fn unix_tight_reset(&mut self, extra_delay: bool) -> Result<(), Error> {
+        info!(
+            "Attempting UnixTight reset with {} delay...",
+            if extra_delay { "extra" } else { "default" }
+        );
 
-            let millis = if extra_delay { 500 } else { 50 };
-            sleep(Duration::from_millis(millis));
+        self.serial.write_dtr_rts(false, false)?;
+        self.serial.write_dtr_rts(true, true)?;
+        self.serial.write_dtr_rts(false, true)?; // IO0 = HIGH, EN = LOW, chip in reset
 
-            self.serial.write_data_terminal_ready(false)?;
-        }
+        sleep(Duration::from_millis(100));
 
-        Ok(())
+        self.serial.write_dtr_rts(true, false)?; // IO0 = LOW, EN = HIGH, chip out of reset
+
+        let millis = if extra_delay { 500 } else { 50 };
+        sleep(Duration::from_millis(millis));
+
+        self.serial.write_dtr_rts(false, false)?; // IO0 = HIGH, done
+        self.serial.write_data_terminal_ready(false)?; // Needed in some environments to ensure IO0 = HIGH
+        self.wait_for_connection()
+    }
+
+    fn usb_jtag_reset(&mut self) -> Result<(), Error> {
```

Comment:
We are also missing the comments here

Reply:
Doesn't really need a doc comment if it's private